### PR TITLE
Remove 'sled-agent' API parameters from 'common'

### PR DESCRIPTION
**Pull request by smklein**
The Sled Agent side of https://github.com/oxidecomputer/omicron/issues/388

### CODE REVIEW

File: nexus-client/src/lib.rs [hunk 1]
```diff
@@ -99,63 +78,22 @@ impl From<omicron_common::api::internal::nexus::InstanceRuntimeState>
...
+            InstanceState::Creating => Self::Creating,
+            InstanceState::Starting => Self::Starting,
+            InstanceState::Running => Self::Running,
+            InstanceState::Stopping => Self::Stopping,
+            InstanceState::Stopped => Self::Stopped,
+            InstanceState::Rebooting => Self::Rebooting,
+            InstanceState::Migrating => Self::Migrating,
+            InstanceState::Repairing => Self::Repairing,
+            InstanceState::Failed => Self::Failed,
+            InstanceState::Destroyed => Self::Destroyed,
```
Comment (david-crespo) [inline nexus-client/src/lib.rs:96]: do we even need these still? I bet there's a way to get rid of them. I'll take a look

**Review by david-crespo — APPROVED** [review]
our beautiful dream, realized!

File: nexus-client/src/lib.rs (hunk 1 above)
Comment (smklein) [reply nexus-client/src/lib.rs:96]: So this PR is admittedly mostly focused on the "pieces of sled agent in `common`" side of things, and hasn't attempted to remove the "pieces of nexus in `common`".

The top-level structure being converted here is `InstanceRuntimeState` (which contains `InstanceState`).

I expected that the dependencies on `omicron_common::api::internal::nexus` to be migrated to either using `nexus::params` or `nexus_client::types` as a separate patch, which would remove the need for the conversion here.

Comment (david-crespo) [reply nexus-client/src/lib.rs:96]: Sounds great!

**Pull request merged**